academic_observatory_workflows/workflows/unpaywall_data_feed_telescope.py

```diff
@@ -0,0 +1,326 @@
+        airflow_vars: List = None,
+    ):
+        """Unpaywall Data Feed telescope.
+
+        :param dag_id: the id of the DAG.
+        :param start_date: the start date of the DAG.
+        :param schedule_interval: the schedule interval of the DAG.
+        :param dataset_id: the dataset id.
+        :param dataset_description: the dataset description.
+        :param merge_partition_field: the BigQuery field used to match partitions for a merge
+        :param bq_merge_days: how often partitions should be merged (every x days)
+        :param schema_folder: the SQL schema path.
+        :param airflow_vars: list of airflow variable keys, for each variable it is checked if it exists in airflow
+        """
+
+        self.validate_schedule_interval_(start_date=start_date, schedule_interval=schedule_interval)
+
+        if airflow_vars is None:
+            airflow_vars = [
+                AirflowVars.DATA_PATH,
+                AirflowVars.PROJECT_ID,
+                AirflowVars.DATA_LOCATION,
+                AirflowVars.DOWNLOAD_BUCKET,
+                AirflowVars.TRANSFORM_BUCKET,
+            ]
+
+        super().__init__(
+            dag_id,
+            start_date,
+            schedule_interval,
+            dataset_id,
+            merge_partition_field,
+            bq_merge_days,
+            schema_folder,
+            dataset_description=dataset_description,
+            batch_load=True,
+            catchup=True,
+            airflow_vars=airflow_vars,
+            airflow_conns=[UnpaywallDataFeedTelescope.AIRFLOW_CONNECTION],
+        )
+
+        self.add_setup_task(self.check_dependencies)
+        self.add_setup_task(self.check_releases)
+
+        self.add_task(self.download)
+        self.add_task(self.upload_downloaded)
+        self.add_task(self.extract)
+        self.add_task(self.transform)
+        self.add_task(self.upload_transformed)
+        self.add_task(self.bq_load_partition)
+
+        self.add_task_chain([self.bq_delete_old, self.bq_append_new, self.cleanup], trigger_rule="none_failed")
+
+    @staticmethod
+    def schedule_days_apart_(
+        *, start_date: pendulum.DateTime, schedule_interval: Union[str, timedelta, relativedelta]
+    ) -> Generator:
+        """Calculate the scheduled days apart.
+
+        :param start_date: DAG start date.
+        :param schedule_interval: DAG schedule interval.
+        :return: A generator that gives back the days apart for each execution.
+        """
+
+        if isinstance(schedule_interval, (timedelta, relativedelta)):
+            while True:
+                yield schedule_interval.days
+
+        a = start_date
+        it = croniter(schedule_interval, start_date)
+
+        while True:
+            b = it.next(datetime)
+            diff = (b - a).days
+            a = b
+            yield diff
+
+    def validate_schedule_interval_(self, *, start_date: pendulum.DateTime, schedule_interval: str):
```

Comment:
Private methods in Python should begin with an underscore, not end with an underscore, e.g. _download_snapshot.